### PR TITLE
chore: Add nightly tests to release branches

### PR DESCRIPTION
**Pull request by jerryharrow**
## Description

Add release branch name pattern to nightly trigger.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by jerryharrow** [comment]
Migrated to OSS from https://github.com/determined-ai/determined-ee/pull/1020

File: .circleci/config.yml [hunk 1]
```diff
@@ -3310,6 +3310,7 @@ workflows:
             branches:
               only:
                 - main
+                - /release-(\d)+(\.(\d)+)+-ee/
```
Comment (MikhailKardash) [inline .circleci/config.yml:3313]: is the `ee` part optional here?

Comment (MikhailKardash) [reply .circleci/config.yml:3313]: Also, would this pipeline trigger upon creation or the release branch, or would it wait until a scheduled time? Seems like we'd want this workflow to happen ASAP.

Comment (jerryharrow) [reply .circleci/config.yml:3313]: You tell me? :-)

Comment (jerryharrow) [reply .circleci/config.yml]: Updated.

Comment (MikhailKardash) [reply .circleci/config.yml]: Upon testing, the `ee` part of regex is mandatory, so probably something like `release-(\d)+(\.(\d)+)+(-ee)?$` instead.

**Review by dannysauer — APPROVED** [review]
We're not using full-blown semver, so this pattern looks good to me.

**Comment by dannysauer** [comment]
@MikhailKardash : would this pipeline trigger upon creation or the release branch, or would it wait until a scheduled time?

It'll fire at a scheduled time on all matching branches, I think.

**Comment by jerryharrow** [comment]
> > @MikhailKardash : would this pipeline trigger upon creation or the release branch, or would it wait until a scheduled time?
> 
> It'll fire at a scheduled time on all matching branches, I think.

Right.   @rcorujo is working with @dannysauer on a different PR that enables developer-requests of nightly to run on branches as well.   It is more complicated than this.

**Pull request merged**